File: jamcha-admin/src/types/article.ts

import { Category } from "./category";
import { Author } from "./author";
import { Tag } from "./tag";
import { Comment } from "./comment";
import { SocialMediaLink } from "./social-media-link";

export interface Article {
  id: number;
  title: string;
  slug: string;
  content: string;
  excerpt: string;
  videoUrl: string;
  featuredImage: string;
  category: Category;
  tags: Tag[];
  comments: Comment[];
  author: Author;
  socialMediaLinks: SocialMediaLink[];
  readingTime: number;
  publishedAt: string; // LocalDateTime as ISO string
  isBreaking: boolean;
  isActive: boolean;
  language: string;
  translations: string;
}

File: jamcha-back/src/main/java/ma/jamcha/jamcha/entities/Article.java
package ma.jamcha.jamcha.entities;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.*;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "articles")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Article {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;

    @Column(unique = true)
    private String slug;

    @Column(columnDefinition = "TEXT")
    private String content;

    @Column(columnDefinition = "TEXT")
    private String excerpt;

    private String featuredImage;
    private String videoUrl;

    @Column(name = "is_active")
    private Boolean isActive = true;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private Category category;

    @ManyToMany
    @JoinTable(
            name = "article_tags",
            joinColumns = @JoinColumn(name = "article_id"),
            inverseJoinColumns = @JoinColumn(name = "tag_id")
    )
    private List<Tag> tags;

    @OneToMany(mappedBy = "article", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Comment> comments;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "author_id", referencedColumnName = "id")
    private Author author;


    private Integer readingTime;

    private LocalDateTime publishedAt;

    @Column(name = "is_breaking")
    private Boolean breaking = false ;

    private String language = "ar";

    @Column(columnDefinition = "json")
    @JdbcTypeCode(SqlTypes.JSON)
    @Basic(fetch = FetchType.LAZY)
    private String translations;
    @OneToMany(mappedBy = "article", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<SocialMediaLink> socialMediaLinks;

}